src/main/java/redis/clients/jedis/DefaultJedisClientConfig.java

```diff
@@ -65,6 +66,21 @@ public String getPassword() {
     return password;
   }
 
+  /**
+   * @param user has to be the same one with which the factory is created
+   * @param password
+   * @throws IllegalArgumentException if the user is not the original user
+   */
+  @Override
+  public synchronized void updatePassword(String user, String password) {
```

Comment:
Is there any reason to use `synchronized` here?

Reply:
@dengliming 

> This doesn't seem to be a very frequent operation.

Exactly. That's why I chose to use simple `synchronized` instead of a lock.

And we need some sort of locking for the safe of password check-and-set operation.